src/ui/ActiveScripts/ActiveScriptsRoot.tsx

```
@@ -3,29 +3,78 @@
  * and provides information about all of the player's scripts that are currently running
  */
 import React, { useState } from "react";
-import Tabs from "@mui/material/Tabs";
-import Tab from "@mui/material/Tab";
+import { Button, Tabs, Tab } from "@mui/material";
 
 import { ActiveScriptsPage } from "./ActiveScriptsPage";
 import { RecentScriptsPage } from "./RecentScriptsPage";
+import { RecentErrorsPage } from "../../ErrorHandling/RecentErrorsPage";
 import { useRerender } from "../React/hooks";
+import { errorModalsAreSuppressed, ErrorState, toggleSuppressErrorModals } from "../../ErrorHandling/ErrorState";
+import { OptionSwitch } from "../React/OptionSwitch";
+import { killAllScripts } from "../../Netscript/killWorkerScript";
+import { SimplePage } from "@enums";
+import { Router } from "../GameRoot";
 
-export function ActiveScriptsRoot(): React.ReactElement {
-  const [tab, setTab] = useState<"active" | "recent">("active");
+type ActiveScriptsTab = SimplePage.ActiveScripts | SimplePage.RecentlyKilledScripts | SimplePage.RecentErrors;
+
+export type ComponentProps = {
+  page: ActiveScriptsTab;
+};
+
+export function ActiveScriptsRoot({ page }: ComponentProps): React.ReactElement {
+  const [tab, setTab] = useState<ActiveScriptsTab>(ErrorState.UnreadErrors > 0 ? SimplePage.RecentErrors : page);
   useRerender(400);
 
-  function handleChange(event: React.SyntheticEvent, tab: "active" | "recent"): void {
+  function handleChange(
+    __event: React.SyntheticEvent,
+    tab: SimplePage.ActiveScripts | SimplePage.RecentlyKilledScripts | SimplePage.RecentErrors,
+  ): void {
     setTab(tab);
+    Router.toPage(tab);
   }
+
```

Bug:
- Suppress error modals.
- Run the test script.
- Switch to the "Active Scripts" tab. The "Recent Errors" tab is shown.
- Check `Router.page()`. It's "Active Scripts".

Wrong `Router.page()` messes up other logic (e.g., `Router.page() !== SimplePage.RecentErrors`).

Potential fix (GitHub UI does not allow me to post this code as a suggestion):
```tsx
  const [tab, setTab] = useState<ActiveScriptsTab>(page);
  useRerender(400);

  useEffect(() => {
    if (ErrorState.UnreadErrors > 0) {
      handleChange(null, SimplePage.RecentErrors);
    }
  }, []);

  function handleChange(
    __event: React.SyntheticEvent | null,
    tab: SimplePage.ActiveScripts | SimplePage.RecentlyKilledScripts | SimplePage.RecentErrors,
  ): void {
    setTab(tab);
    Router.toPage(tab);
  }
```

---

This suggested changed has seemed to fix the issue. The route is now properly RecentErrors when the Active Scripts tab is clicked and there are unread errors.